### DISCUSSION TITLE
Is Qdrant Query Speed Inconsistency Due to Caching?

### DISCUSSION THREAD

**Discussion by dylanewbie** [qdrant/qdrant#4540] [General]
I would like to know if Qdrant has a caching mechanism.

I have a collection with 9 million points and I am using a Rust program with an HTTP client to call the Qdrant API for queries. The program receives a CSV file containing 10,000 points to query.

I noticed that the first query takes 90 seconds (excluding CSV loading time). For the same data, subsequent queries take only 20 seconds. However, when I switch to new data, the first query is slow again, but subsequent queries are fast.

Does Qdrant have a caching mechanism? How can I temporarily disable caching? I need to measure the average query time, but currently, I can only get accurate measurements by restarting the system, as the times after the first query do not seem reliable.

Thank you!

**Comment by generall** [comment]
hey @dylanewbie, if you are using any kind of on-disk option for vectors or indexes, the first request indeed will hit the cold disk cache.
